### PR TITLE
KAFKA-15085: Make Timer.java implement AutoCloseable

### PR DESCRIPTION
**Pull request by joobisb**
As a best practice implement AutoCloseable which many automatic bug finders will flag a warning if an object of a class is marked as AutoCloseable but is not closed properly in the code.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: server-common/src/main/java/org/apache/kafka/server/util/timer/SystemTimer.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ public int size() {
         return taskCounter.get();
     }
 
-    public void shutdown() {
+    public void close() {
```
Comment (divijvaidya) [inline server-common/src/main/java/org/apache/kafka/server/util/timer/SystemTimer.java:110]: nit

please add `@Override`

**Review by divijvaidya — COMMENTED** [review]
Looks good except for a small comment. Let's wit for @dajac's review on this one since he expressed interest in this PR.

**Comment by dajac** [comment]
@divijvaidya Don't wait on me cause I may not have time before the end of the week.

**Comment by divijvaidya** [comment]
Unrelated test failures
```
[Build / JDK 17 and Scala 2.13 / integration.kafka.server.FetchFromFollowerIntegrationTest.testRackAwareRangeAssignor()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13872/1/testReport/junit/integration.kafka.server/FetchFromFollowerIntegrationTest/Build___JDK_17_and_Scala_2_13___testRackAwareRangeAssignor__/)
[Build / JDK 8 and Scala 2.12 / org.apache.kafka.connect.mirror.integration.MirrorConnectorsIntegrationTransactionsTest.testOffsetTranslationBehindReplicationFlow()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13872/1/testReport/junit/org.apache.kafka.connect.mirror.integration/MirrorConnectorsIntegrationTransactionsTest/Build___JDK_8_and_Scala_2_12___testOffsetTranslationBehindReplicationFlow__/)
[Build / JDK 8 and Scala 2.12 / integration.kafka.server.FetchFromFollowerIntegrationTest.testRackAwareRangeAssignor()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13872/1/testReport/junit/integration.kafka.server/FetchFromFollowerIntegrationTest/Build___JDK_8_and_Scala_2_12___testRackAwareRangeAssignor__/)
[Build / JDK 8 and Scala 2.12 / org.apache.kafka.controller.QuorumControllerTest.testBalancePartitionLeaders()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13872/1/testReport/junit/org.apache.kafka.controller/QuorumControllerTest/Build___JDK_8_and_Scala_2_12___testBalancePartitionLeaders__/)
[Build / JDK 11 and Scala 2.13 / integration.kafka.server.FetchFromFollowerIntegrationTest.testRackAwareRangeAssignor()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13872/1/testReport/junit/integration.kafka.server/FetchFromFollowerIntegrationTest/Build___JDK_11_and_Scala_2_13___testRackAwareRangeAssignor__/)
[Build / JDK 11 and Scala 2.13 / kafka.api.ConsumerBounceTest.testCloseDuringRebalance()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13872/1/testReport/junit/kafka.api/ConsumerBounceTest/Build___JDK_11_and_Scala_2_13___testCloseDuringRebalance__/)
[Build / JDK 11 and Scala 2.13 / org.apache.kafka.controller.QuorumControllerTest.testBalancePartitionLeaders()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13872/1/testReport/junit/org.apache.kafka.controller/QuorumControllerTest/Build___JDK_11_and_Scala_2_13___testBalancePartitionLeaders__/)
```

@joobisb if you can address the nit above, we can continue to merge this one in.

**Comment by joobisb** [comment]
@divijvaidya i've addressed the comment, please have a look

**Pull request merged**